app/uk/gov/hmrc/economiccrimelevyregistration/controllers/AmendRegistrationStartController.scala

```diff
@@ -61,6 +70,22 @@ class AmendRegistrationStartController @Inject() (
       }
   }
 
+  private def routeToAmendReason(eclRegistrationReference: String, internalId: String)(implicit
+    hc: HeaderCarrier
+  ): Future[Result] = {
+    val result = for {
+      registration            <- registrationService.getOrCreateRegistration(internalId)
+      getSubscriptionResponse <- registrationService.getSubscription(eclRegistrationReference)
+      upsertedRegistration    <-
+        registrationService.upsertRegistration(
+          registrationService.transformToRegistration(registration, getSubscriptionResponse)
+        )
+    } yield upsertedRegistration
+
+    result.map(_ => Redirect(routes.AmendReasonController.onPageLoad(CheckMode).url))
```

Comment:
we're not handling when an error happens

Reply:
Done